### PR TITLE
Feature/split up to separate modules

### PR DESCRIPTION
**Pull request by royteeuwen**

### CODE REVIEW

File: integration-tests/src/main/features/opentelemetry-java.json [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      "id": "${project.groupId}:opentelemetry-exporter-otlp:${opentelemetry.java.version}.000-SNAPSHOT",
+      "start-order": 5
+    },
+    {
+      "id": "${project.groupId}:opentelemetry-exporter-otlp-common:${opentelemetry.java.version}.000-SNAPSHOT",
+      "start-order": 5
+    },
+    {
+      "id": "com.fasterxml.jackson.core:jackson-core:2.17.2",
+      "start-order": 5
```
Comment (bdhoine) [inline integration-tests/src/main/features/opentelemetry-java.json:66]: should thse all have start order 5?

File: integration-tests/pom.xml [hunk 2]
```diff
@@ -190,14 +206,6 @@
                     </execution>
                 </executions>
             </plugin>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-compiler-plugin</artifactId>
-                <configuration>
-                    <source>11</source>
```
Comment (bdhoine) [inline integration-tests/pom.xml:197]: should we specify a java version or does it inherit this config?

File: integration-tests/src/test/java/be/orbinson/osgi/opentelemetry/it/OpenTelemetryIT.java [hunk 3]
```diff
@@ -18,6 +18,14 @@ public class OpenTelemetryIT {
 
     private static final int SLING_PORT = Integer.getInteger("HTTP_PORT", 8080);
 
+    @Test
+    public void testOsgi() throws InterruptedException {
+        System.out.println("http://localhost:" + SLING_PORT + "/system/console");
+        if (System.getProperty("debug.osgi") != null) {
+            Thread.sleep(1000000);
```
Comment (bdhoine) [inline integration-tests/src/test/java/be/orbinson/osgi/opentelemetry/it/OpenTelemetryIT.java:25]: is this for local debugging purposes?

File: integration-tests/src/test/java/be/orbinson/osgi/opentelemetry/it/OpenTelemetryIT.java [hunk 4]
```diff
@@ -31,8 +39,25 @@ public void testOpenTelemetryBundlesActive() throws Exception {
         HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
 
         String responseBody = response.body();
-        assertEquals("Active", ((List<Map<String, Object>>) JsonPath.read(responseBody, "$.data[?(@.name=='opentelemetry-java')]")).get(0).get("state"));
-        assertEquals("Active", ((List<Map<String, Object>>) JsonPath.read(responseBody, "$.data[?(@.name=='opentelemetry-java-instrumentation')]")).get(0).get("state"));
+        assertEquals("Active", ((List<Map<String, Object>>) JsonPath.read(responseBody, "$.data[?(@.name=='opentelemetry-api')]")).get(0).get("state"));
```
Comment (bdhoine) [inline integration-tests/src/test/java/be/orbinson/osgi/opentelemetry/it/OpenTelemetryIT.java]: might use a helper method for readability

File: opentelemetry-java-instrumentation/opentelemetry-instrumentation-annotations/pom.xml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    <properties>
+        <opentelemetry.groupId>${opentelemetry.java.instrumentation.groupId}</opentelemetry.groupId>
+        <opentelemetry.version>${opentelemetry.java.instrumentation.version}</opentelemetry.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>${opentelemetry.groupId}</groupId>
+            <artifactId>opentelemetry-instrumentation-annotations</artifactId>
```
Comment (bdhoine) [inline opentelemetry-java-instrumentation/opentelemetry-instrumentation-annotations/pom.xml]: `${project.artifactId}`?

File: opentelemetry-java/opentelemetry-exporter-common/pom.xml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        <plugins>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>5.1.9</version>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <!-- We are using okhttp as sender, which does not have a dependency on grpc -->
+                        <Import-Package>io.grpc*;resolution:=optional,*</Import-Package>
```
Comment (bdhoine) [inline opentelemetry-java/opentelemetry-exporter-common/pom.xml]: `io.grpc.*`?

**Review by bdhoine — COMMENTED** [review]
some minor comments/questions

File: integration-tests/pom.xml (hunk 2 above)
Comment (royteeuwen) [reply integration-tests/pom.xml:197]: this indeed inherits

File: integration-tests/src/main/features/opentelemetry-java.json (hunk 1 above)
Comment (royteeuwen) [reply integration-tests/src/main/features/opentelemetry-java.json:66]: they are all required for opentelemetry to work, so i think that it's indeed best to set it to 5

File: integration-tests/src/test/java/be/orbinson/osgi/opentelemetry/it/OpenTelemetryIT.java (hunk 3 above)
Comment (royteeuwen) [reply integration-tests/src/test/java/be/orbinson/osgi/opentelemetry/it/OpenTelemetryIT.java:25]: yep!

File: integration-tests/src/test/java/be/orbinson/osgi/opentelemetry/it/OpenTelemetryIT.java (hunk 4 above)
Comment (royteeuwen) [reply integration-tests/src/test/java/be/orbinson/osgi/opentelemetry/it/OpenTelemetryIT.java]: done

File: opentelemetry-java/opentelemetry-exporter-common/pom.xml (hunk 6 above)
Comment (royteeuwen) [reply opentelemetry-java/opentelemetry-exporter-common/pom.xml]: no, on purpose, we want to also exclude io.grpc package itself

File: opentelemetry-java-instrumentation/opentelemetry-instrumentation-annotations/pom.xml (hunk 5 above)
Comment (royteeuwen) [reply opentelemetry-java-instrumentation/opentelemetry-instrumentation-annotations/pom.xml]: done

**Pull request closed without merging**